pandas/tseries/plotting.py

```diff
@@ -278,8 +280,22 @@ def _maybe_convert_index(ax, data):
 # Patch methods for subplot. Only format_dateaxis is currently used.
 # Do we need the rest for convenience?
 
-
-def format_dateaxis(subplot, freq):
+def timeTicks(x, pos, n_decimals):
+    ''' Convert seconds to 'D days HH:MM:SS.F' '''
+    s, ns = divmod(x, 1e9)
+    m, s = divmod(s, 60)
+    h, m = divmod(m, 60)
+    d, h = divmod(h, 24)
+    decimals = int(ns * 10**(n_decimals - 9))
+    s = r'{:02d}:{:02d}:{:02d}'.format(int(h), int(m), int(s))
+    if n_decimals > 0:
```

Comment:
Maybe the

```
Components(days=0, hours=0, minutes=2, seconds=0, milliseconds=500, microseconds=0, nanoseconds=0)
```

could compress the code a bit, but I'm not sure it would save significant space over what I've already got.